packages/edit-site/src/components/global-styles/custom-css.js

```diff
@@ -37,6 +41,15 @@ function CustomCSSControl() {
 			return;
 		}
 		setCustomCSS( value );
+		// TODO: transform it to find syntax error. Note that it will be transformed again
+		// after applied to the preview frame. We should refactor this to prevent double-transform.
+		const [ transformed ] = transformStyles(
+			[ { css: value } ],
+			'.editor-styles-wrapper'
+		);
+		setCSSError(
+			transformed === null ? __( 'Error while parsing the CSS.' ) : null
+		);
```

Comment:
Maybe we should move the error handling `onBlur` event? How helpful is the error message while a user is typing?

Reply:
Yeh, while typing there will be lots of invalid states. I have moved it to onBlur, and only runs in onChange if there was an existing error so that it gets cleared once the user fixes the problem.